Source/Core/DolphinWX/GameListCtrl.cpp

```diff
@@ -455,9 +462,15 @@ void CGameListCtrl::InsertItemInReportView(long _Index)
 			if (!std::getline(titlestxt, line) && titlestxt.eof())
 				break;
 
-			if (line.substr(0,rISOFile.GetUniqueID().size()) == rISOFile.GetUniqueID())
+			std::string game_id = rISOFile.GetUniqueID();
+
+			// Ignore publisher ID for WAD files
+			if (rISOFile.GetPlatform() == GameListItem::WII_WAD)
+				game_id.erase(game_id.size() - 2);
+
+			if (line.substr(0, line.find("=") - 1) == game_id)
```

Comment:
This can be changed from:

```cpp
if (line.substr(0, line.find("=") - 1) == game_id)
{
	name = line.substr(line.find("=") + 2);
	break;
}
```
to

```cpp
const size_t equals_index = line.find('=');
if (line.substr(0, equals_index - 1) == game_id)
{
	name = line.substr(equals_index + 2);
	break;
}
```

so you aren't doing a string look-up twice.